### ISSUE TITLE
[accessibility] Interactive markers should be `role="button"`

### ISSUE THREAD

**Issue by Malvoz** [maplibre/maplibre-gl-js#356]
Interactive markers (e.g. markers that bind to popups) are focusable and actionable elements, and should have appropriate semantics (i.e. [`role="button"`](https://www.w3.org/TR/wai-aria/#button)).

Related WCAG 2.1 Success Criterion: [4.1.2 Name, Role, Value](https://www.w3.org/TR/WCAG21/#name-role-value) (this issue is to address the <q>Role</q> part of the Success Criterion for interactive markers).

<hr>

(The same issue was also reported with MapBox: https://github.com/mapbox/mapbox-gl-js/issues/11031.)

**Comment by skorasaurus** [comment]
Hi @Malvoz : 

Even though this issue was closed for inactivity; interactive markers do now have the button role applied; this was implemented in https://github.com/maplibre/maplibre-gl-js/pull/6517 which appears in MapLibre 5.9.0. courtesy of @cmburcus (thank you!) 

You can see an example of this at https://maplibre.org/maplibre-gl-js/docs/examples/attach-a-popup-to-a-marker-instance/

**Comment by andrewharvey** [comment]
hmm per my comment at https://github.com/maplibre/maplibre-gl-js/pull/6517#issuecomment-4714794106 #6517 appears to implement `role=button` for any marker, even non interactive which seems wrong.
